src/mame/drivers/aa310.cpp

```diff
@@ -1076,12 +1087,11 @@ void aa500_state::aa500(machine_config &config)
 	m_ioc->peripheral_w<5>().set(FUNC(aa500_state::peripheral5_w));
 	m_ioc->peripheral_r<6>().set_constant(0xff);
 	m_ioc->peripheral_w<6>().set(FUNC(aa500_state::peripheral6_w));
-	m_ioc->gpio_r<0>().set("i2cmem", FUNC(i2c_pcf8570_device::read_sda));
-	m_ioc->gpio_r<0>().append("rtc", FUNC(pcf8573_device::sda_r));
-	m_ioc->gpio_w<0>().set("i2cmem", FUNC(i2c_pcf8570_device::write_sda));
-	m_ioc->gpio_w<0>().append("rtc", FUNC(pcf8573_device::sda_w));
-	m_ioc->gpio_w<1>().set("i2cmem", FUNC(i2c_pcf8570_device::write_scl));
-	m_ioc->gpio_w<1>().append("rtc", FUNC(pcf8573_device::scl_w));
+	m_ioc->gpio_r<0>().set([this]() { return m_i2cmem->read_sda() & m_rtc->sda_r(); });
+	m_ioc->gpio_w<0>().set(m_i2cmem, FUNC(i2c_pcf8570_device::write_sda));
+	m_ioc->gpio_w<0>().append(m_rtc, FUNC(pcf8573_device::sda_w));
+	m_ioc->gpio_w<1>().set(m_i2cmem, FUNC(i2c_pcf8570_device::write_scl));
+	m_ioc->gpio_w<1>().append(m_rtc, FUNC(pcf8573_device::scl_w));
```

Comment:
It may not matter in this case, but in real-world implementations, the targets should also see when SDA and/or SCL is being pulled low by another target, and controllers need to be able to see when SCL is being pulled low by any of the targets.  This is particularly important when targets that implement clock stretching are present.

The standard does say that controllers aren’t required to support clock stretching in situations where it’s known that none of targets implement it, though (UM10204, table 3, footnote 2).